CMakeLists.txt

```diff
@@ -0,0 +1,228 @@
+include_directories(${OPENGL_INCLUDE_DIR})
+list(APPEND XOREOS_LIBRARIES ${OPENGL_LIBRARIES})
+
+find_package(Freetype REQUIRED)
+include_directories(${FREETYPE_INCLUDE_DIRS})
+link_directories(${FREETYPE_LIBRARY_DIRS})
+list(APPEND XOREOS_LIBRARIES ${FREETYPE_LIBRARIES})
+
+find_package(Mad REQUIRED)
+include_directories(${MAD_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${MAD_LIBRARIES})
+
+find_package(Faad REQUIRED)
+include_directories(${FAAD_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${FAAD_LIBRARIES})
+
+find_package(Vorbis REQUIRED)
+include_directories(${VORBIS_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${VORBIS_LIBRARIES})
+
+find_package(XviD REQUIRED)
+include_directories(${XVID_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${XVID_LIBRARIES})
+
+
+find_package(GLEW)
+if(NOT GLEW_FOUND)
+  add_definitions(-DGLEW_STATIC)
+  set(GLEW_LIBRARIES    glew)
+  set(GLEW_INCLUDE_DIRS glew)
+
+  parse_configure(configure.ac glew)
+endif()
+include_directories(${GLEW_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${GLEW_LIBRARIES})
+
+
+find_package(Lua51)
+if(NOT LUA51_FOUND)
+  set(LUA_LIBRARIES   lua)
+  set(LUA_INCLUDE_DIR lua)
+
+  parse_configure(configure.ac lua)
+endif()
+include_directories(${LUA_INCLUDE_DIR})
+list(APPEND XOREOS_LIBRARIES ${LUA_LIBRARIES})
+
+
+# -------------------------------------------------------------------------
+# xoreos main targets, parsed from configure.ac and */Makefile.am
+include_directories(${PROJECT_SOURCE_DIR} ${PROJECT_SOURCE_DIR}/src)
+add_definitions(-DPACKAGE_STRING="xoreos ${xoreos_VERSION}")
+parse_configure(configure.ac src)
+target_link_libraries(xoreos ${XOREOS_LIBRARIES})
+
+# -------------------------------------------------------------------------
+# documentation
+option(XOREOS_BUILD_DOCUMENTATION "Use Doxygen to create the HTML based API documentation" OFF)
+if(XOREOS_BUILD_DOCUMENTATION)
+  find_package(Doxygen)
+  if (NOT DOXYGEN_FOUND)
+    message(FATAL_ERROR "Doxygen is needed to build the documentation. Please install it correctly")
+  endif()
+
+  add_custom_target(doxygen
+    COMMAND ${DOXYGEN_EXECUTABLE}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    SOURCES ${PROJECT_SOURCE_DIR}/Doxyfile
+  )
+  add_custom_target(doxygen-clean
+    COMMAND rm -rf doc/doxygen/html/
+    COMMAND rm -rf doc/doxygen/latex/
+    COMMAND rm -rf doc/doxygen/man/
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+  )
+endif()
+
+
+# -------------------------------------------------------------------------
+# packaging
+# see: http://www.vtk.org/Wiki/CMake:CPackPackageGenerators
+set(CPACK_GENERATOR "TGZ")
+
+set(CPACK_PACKAGE_NAME    ${PROJECT_NAME})
+set(CPACK_PACKAGE_VERSION ${xoreos_VERSION})
+set(CPACK_PACKAGE_CONTACT "Sven Hesse <drmccoy@drmccoy.de>")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Xoreos Engine: A new implementation of BioWare's Aurora engine.")
+set(CPACK_PACKAGE_DESCRIPTION "Xoreos Engine: A new implementation of BioWare's Aurora engine.")
+set(CPACK_PACKAGE_VENDOR "Xoreos Project")
```

Comment:
If it doesn't stand out too much, I'd rather like "xoreos" to always be all-lowercase. The only place were we don't do that is the Mac Bundle name thing, because I've been told it looks bad otherwise.